### PR TITLE
Clean up chat code

### PR DESCRIPTION
**Pull request by hakubo**
This branch cleans up chat code a bit and fixes some minor issues. 
Left to do:
- [ ] Testing
- [ ] CR

### CODE REVIEW

File: extensions/wikia/Chat2/js/ChatWidget.js [hunk 1]
```diff
@@ -96,11 +96,11 @@ var ChatWidget = {
 		});
 
 		$('.chatCarousel').each(function () {
-			$(this).get(0).innerHTML = output;
+			$(this).get(0).html(output);
```
Comment (hakubo) [inline extensions/wikia/Chat2/js/ChatWidget.js:99]: There is no innerHTML property on jquery element

File: extensions/wikia/Chat2/js/models/models.js [hunk 2]
```diff
@@ -1,41 +1,36 @@
-var STATUS_STATE_PRESENT = 'here'; // strings instead of ints just for easier debugging. always use the vars, don't hardcode strings w/these states elsewhere.
-var STATUS_STATE_AWAY = 'away';
+window.STATUS_STATE_PRESENT = 'here'; // strings instead of ints just for easier debugging. always use the vars, don't hardcode strings w/these states elsewhere.
```
Comment (hakubo) [inline extensions/wikia/Chat2/js/models/models.js:1]: I think it shows better that these are globals

File: extensions/wikia/Chat2/js/ChatWidget.js [hunk 3]
```diff
@@ -148,26 +148,26 @@ var ChatWidget = {
...
 	 */
-	processModuleTemplate: function ($t) {
-		ChatWidget.initCarousel($t.find('.chat-whos-here'));
+	processModuleTemplate: function ($element) {
+		ChatWidget.initCarousel($element.find('.chat-whos-here'));
 
 		// process i18n the messages
-		$t.find('[data-msg-id]').each(function () {
-			var $e = $(this);
-			$e.text($.msg($e.data('msg-id'), $e.data('msg-param')));
```
Comment (nandy-andy) [inline extensions/wikia/Chat2/js/ChatWidget.js:159]: I see what was going on here! Elemen**t** and modul**e** -> `$t` and `$e` :)

File: extensions/wikia/Chat2/js/controllers/ChatBanModalLogs.js [hunk 4]
```diff
@@ -1,29 +1,33 @@
...
-					}
-				);
-			},
-			chatBanModal = new ChatBanModal(title, okCallback, {'isChangeBan': true, userId: userId});
+		var userId = $(this).data('user-id');
+
+		require(['chat-ban-modal'], function (chatBanModal) {
+			chatBanModal.open(
+				$.msg('chat-ban-modal-change-ban-heading'),
+				function (expires, reason) {
```
Comment (nandy-andy) [inline extensions/wikia/Chat2/js/controllers/ChatBanModalLogs.js:13]: Minor comment: only in favor of readability I'd consider extracting the callback function.

File: extensions/wikia/Chat2/js/controllers/ChatBanModalLogs.js [hunk 5]
```diff
@@ -1,29 +1,33 @@
...
+					$.post(window.wgScript + '?action=ajax&rs=ChatAjax&method=blockOrBanChat', {
+							userToBanId: userId,
+							time: expires,
+							reason: reason,
+							mode: 'global',
+							token: mw.user.tokens.get('editToken')
+						},
+						function () {
```
Comment (nandy-andy) [inline extensions/wikia/Chat2/js/controllers/ChatBanModalLogs.js:21]: Do we need the anonymous function here?

**Review by nandy-andy — APPROVED** [review]
I left minor comments and a question out of curiosity ;) Overall: LGTM :+1:

**Pull request closed without merging**